src/Engine/ProtoCore/Utils/ClassUtils.cs

```diff
@@ -51,8 +51,33 @@ public static int GetUpcastCountTo(ClassNode from, ClassNode to, RuntimeCore run
                 return int.MaxValue;
 
             return upcastChain.IndexOf(toID);
+        }
 
+        // TODO_MSIL: replace with the existing ClassNode signature of this function when ClassNode are properly populated in MSIL.
+        internal static int GetUpcastCountTo((int id, System.Type type) from, (int id, System.Type type) target)
```

Comment:
Why do we need to replace with `ClassNode` if this implementation works?

Reply:
To re-use the pre-existing method (if possible)
Not sure if the ClassNode can do anything more (represent something more) than System.Type can.
Maybe pure DesignScript classes ? not sure if we can define classes in design script...And if we can...maybe we can simply create System.Type instead of a ClassNode, but since the ClassNode already exists...we might as well use it

Reply:
Yes, we have builtin DesignScript classes to implement some built-in nodes. See BuiltIn.ds and FunctionObject.ds. FunctionObject.ds is mainly to create partial functions and support nodes that accept function pointers and we might be doing away with those so it may not be useful to continue supporting it. BuiltIn.ds has some `List` nodes that are implemented in DesignScript. These might need to be ported to C# functions.

Reply:
Well I guess we can keep this method then (I will remove the TODO)
But I am not sure if we can/should entirely get rid of the ClassNode (and simply replace it with System.Type ).
I guess we can figure that out later